relay_test.go

```diff
@@ -89,17 +89,23 @@ func TestRelay(t *testing.T) {
 	})
 }
 
-func DisabledTestRelayHandlesCrashedPeers(t *testing.T) {
-	withRelayedEcho(t, func(_, server, client *Channel, ts *testutils.TestServer) {
-		ctx, cancel := NewContext(time.Second)
+func TestRelayHandlesClosedPeers(t *testing.T) {
+	opts := serviceNameOpts("test").SetRelayOnly().
+		// The relay will fail to connect when the server is closed.
+		AddLogFilter("Failed to connect to relay host", 1)
+	testutils.WithTestServer(t, opts, func(ts *testutils.TestServer) {
+		ctx, cancel := NewContext(300 * time.Millisecond)
 		defer cancel()
 
+		testutils.RegisterEcho(ts.Server(), nil)
+		client := ts.NewClient(serviceNameOpts("client"))
+		client.Peers().Add(ts.HostPort())
+
 		sc := client.GetSubChannel("test")
 		_, _, _, err := raw.CallSC(ctx, sc, "echo", []byte("fake-header"), []byte("fake-body"))
 		require.NoError(t, err, "Relayed call failed.")
 
-		// Simulate a server crash.
-		server.Close()
+		ts.Server().Close()
```

Comment:
In this case, we're closing the server before the test ends, since we have more logic to execute below (the `CallSC`). If we used defer, the server wouldn't close until the end of the test function.

defer is useful for cleanup, but in this case, the `Close` is actually important to what's being tested